### PR TITLE
[receiver/purefa]  Add validate on config components

### PR DESCRIPTION
**Pull request by dgoscn**
**Description:**
There were missing code snippet from validate section on config.go inside the purefa receiver. At this piece of code, a block is inserted and the following unit test.

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14886

**Testing:**
Same present on **README.md** section

**Documentation:**
Still in progress

### CODE REVIEW

**Comment by atoulme** [comment]
You might want to add a unit test covering validation checks.

File: receiver/purefareceiver/config.go [hunk 1]
```diff
@@ -61,6 +63,23 @@ type ReloadIntervals struct {
...
+	}
+	if len(c.Settings.ReloadIntervals.Directories.String()) == 0 {
+		err = multierr.Append(err, errors.New("Directories not provided and is required"))
+	}
+	if len(c.Settings.ReloadIntervals.Pods.String()) == 0 {
+		err = multierr.Append(err, errors.New("Pods not provided and is required"))
+	}
+	if len(c.Settings.ReloadIntervals.Volumes.String()) == 0 {
+		err = multierr.Append(err, errors.New("Volumes not provided and is required"))
+	}
```
Comment (bogdandrutu) [inline receiver/purefareceiver/config.go:82]: Personal taste, is to compare with empty string for strings, and with len 0 for slice/map.

Comment (bogdandrutu) [inline receiver/purefareceiver/config.go:82]:
```suggestion
	if c.Settings.ReloadIntervals.Array.String() == "" {
		err = multierr.Append(err, errors.New("Arrays not provided and is required"))
	}
	if c.Settings.ReloadIntervals.Host.String() == "" {
		err = multierr.Append(err, errors.New("Hosts not provided and is required"))
	}
	if c.Settings.ReloadIntervals.Directories.String() == "" {
		err = multierr.Append(err, errors.New("Directories not provided and is required"))
	}
	if c.Settings.ReloadIntervals.Pods.String() == "" {
		err = multierr.Append(err, errors.New("Pods not provided and is required"))
	}
	if c.Settings.ReloadIntervals.Volumes.String() == "" {
		err = multierr.Append(err, errors.New("Volumes not provided and is required"))
	}
```

**Review by bogdandrutu — APPROVED** [review]

File: receiver/purefareceiver/config.go (hunk 1 above)
Comment (dgoscn) [reply receiver/purefareceiver/config.go]: @bogdandrutu thanks for this review. I agree with this

Comment (jpkrohling) [reply receiver/purefareceiver/config.go:85]: This is similar to the Validate part of the purefb receiver. Once it's aligned with that, I'll merge this.

File: receiver/purefareceiver/config.go [hunk 2]
```diff
@@ -68,19 +68,19 @@ type ReloadIntervals struct {
...
-	if c.Settings.ReloadIntervals.Directories.String() == "" {
+	if len(c.Settings.ReloadIntervals.Directories.String()) == 0 {
 		err = multierr.Append(err, errors.New("Directories not provided and is required"))
 	}
-	if c.Settings.ReloadIntervals.Pods.String() == "" {
+	if len(c.Settings.ReloadIntervals.Pods.String()) == 0 {
 		err = multierr.Append(err, errors.New("Pods not provided and is required"))
 	}
-	if c.Settings.ReloadIntervals.Volumes.String() == "" {
+	if len(c.Settings.ReloadIntervals.Volumes.String()) == 0 {
```
Comment (bogdandrutu) [inline receiver/purefareceiver/config.go:83]: You reverted my changes....

**Review by bogdandrutu — DISMISSED** [review]

**Review by jpkrohling — DISMISSED** [review]

**Comment by jpkrohling** [comment]
Looks like the go.mod/sum are outdated now.

**Review by jpkrohling — APPROVED** [review]

File: receiver/purefareceiver/go.mod [hunk 3]
```diff
@@ -3,15 +3,15 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/receiver/purefa
 go 1.18
 
 require (
-	github.com/open-telemetry/opentelemetry-collector-contrib/extension/bearertokenauthextension v0.69.0
-	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver v0.69.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/extension/bearertokenauthextension v0.70.0
```
Comment (jpkrohling) [inline receiver/purefareceiver/go.mod:6]: This change is still bugging me, as the current main is already at v0.70.0. Can you please rebase this PR? I'm not expecting any changes to the go.mod/sum as part of this PR.

**Review by MovieStoreGuy — COMMENTED** [review]
So far this config only checks the `Config.Settings`, should it also check the other exposed files as well?

**Comment by MovieStoreGuy** [comment]
Hey @dgoscn,

Are you still working on this change?

**Comment by dgoscn** [comment]
> Hey @dgoscn,
> 
> Are you still working on this change?

Hi @MovieStoreGuy. Thanks for your review. Yes, I am working on this. I will push de changes asap. 

Thank you

File: receiver/purefareceiver/config.go [hunk 4]
```diff
@@ -64,6 +66,23 @@ type ReloadIntervals struct {
 }
 
 func (c *Config) Validate() error {
-	// TODO(dgoscn): perform config validation
-	return nil
+	var errs error
+
+	if c.Settings.ReloadIntervals.Array.String() == "" {
```
Comment (jpkrohling) [inline receiver/purefareceiver/config.go:71]: This seems to have been reverted again. Please, use what's being used for purefb.

**Review by jpkrohling — CHANGES_REQUESTED** [review]

File: receiver/purefareceiver/config.go (hunk 4 above)
Comment (dgoscn) [reply receiver/purefareceiver/config.go:71]: Sure @jpkrohling

**Review by jpkrohling — APPROVED** [review]

**Comment by MovieStoreGuy** [comment]
Hey @dgoscn , 

Could I ask you to rebase with main please?

**Comment by dgoscn** [comment]
> Hey @dgoscn ,
> 
> Could I ask you to rebase with main please?

Sure @MovieStoreGuy thansk for the review

**Comment by jpkrohling** [comment]
The go.mod/sum files are still outdated.

**Review by MovieStoreGuy — APPROVED** [review]

**Comment by MovieStoreGuy** [comment]
Please rebase.

**Comment by jpkrohling** [comment]
Once this is rebased, I'll review and merge this.

**Comment by dgoscn** [comment]
Hi, @jpkrohling. Thank you for your last comment. Can you please check my previous commit?

Thank you

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**